Add PIM PAG API documentation in graph beta (#19585)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/privilegedaccessgroupassignmentscheduleinstance-filterbycurrentuser.md

---
title: "privilegedAccessGroupAssignmentScheduleInstance: filterByCurrentUser"

> description: "Retrieve the schedule instances of privileged access to a group relationship assignment for the calling principal."

Reviewer: ```suggestion
description: "In PIM for groups, retrieve the schedule instances for access assignments for the calling principal to groups that are governed by PIM."
```

author: "mandardalvi"
ms.localizationpriority: medium
ms.prod: "governance"
doc_type: apiPageType
---

# privilegedAccessGroupAssignmentScheduleInstance: filterByCurrentUser
Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Retrieve the schedule instances of privileged access to a group relationship assignment for the calling principal.

Reviewer: Match with Line 3

## Permissions
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type|Permissions (from least to most privileged)|
|:---|:---|

> |Delegated (work or school account)|PrivilegedAccess.ReadWrite.AzureADGroup, PrivilegedAccess.Read.AzureADGroup|

Reviewer: Proper order, please.

|Delegated (personal Microsoft account)|Not supported.|
|Application|PrivilegedAccess.ReadWrite.AzureADGroup, PrivilegedAccess.Read.AzureADGroup|

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http

> GET /identityGovernance/privilegedAccess/group/assignmentScheduleInstances/filterByCurrentUser

Reviewer: I believe the function parameters are required. Apply that to this URL.

```

## Function parameters
In the request URL, provide the following query parameters with values.

> The following table shows the parameters that can be used with this function.

Reviewer: ```suggestion
The following table shows the parameters that are required with this function.
```

|Parameter|Type|Description|
|:---|:---|:---|
|on|assignmentScheduleInstanceFilterByCurrentUserOptions|Filter used to query assignmentScheduleInstances. The possible values are `principal`, `unknownFutureValue`. Required.|

## Request headers
|Name|Description|
|:---|:---|
|Authorization|Bearer {token}. Required.|

## Request body
Do not supply a request body for this method.

## Response

If successful, this function returns a `200 OK` response code and a [privilegedAccessGroupAssignmentScheduleInstance](../resources/privilegedaccessgroupassignmentscheduleinstance.md) collection in the response body.

## Examples

### Request
The following is an example of a request.
<!-- {
  "blockType": "request",
  "name": "privilegedaccessgroupassignmentscheduleinstancethis.filterbycurrentuser"
}
-->
``` http

> GET https://graph.microsoft.com/beta/identityGovernance/privilegedAccess/group/assignmentScheduleInstances/filterByCurrentUser(on='parameterValue')

Reviewer: Replace 'parameterValue' with 'principal'

```

### Response
The following is an example of the response.
>**Note:** The response object shown here might be shortened for readability.
<!-- {
  "blockType": "response",
  "truncated": true,
  "@odata.type": "Collection(microsoft.graph.privilegedAccessGroupAssignmentScheduleInstance)"
}
-->
``` http
HTTP/1.1 200 OK
Content-Type: application/json

> {

Reviewer: Real payload, please!
